app/controllers/v0/sign_in_controller.rb

```diff
@@ -175,31 +176,38 @@ def revoke_all_sessions
 
     def logout # rubocop:disable Metrics/MethodLength
       client_id = params[:client_id].presence
-      anti_csrf_token = anti_csrf_token_param.presence
-
-      if client_config(client_id).blank?
+      if client_id && client_config(client_id).blank?
         raise SignIn::Errors::MalformedParamsError.new message: 'Client id is not valid'
       end
 
-      unless load_user(skip_expiration_check: true)
-        raise SignIn::Errors::LogoutAuthorizationError.new message: 'Unable to Authorize User'
-      end
+      user_verification = nil
 
-      SignIn::SessionRevoker.new(access_token: @access_token, anti_csrf_token:).perform
-      delete_cookies if token_cookies
+      if (session = SignIn::OAuthSession.find_by(handle: @access_token&.session_handle))
+        user_verification = session.user_verification
+        client_id = session.client_id if client_id.blank?
+        anti_csrf_token = anti_csrf_token_param.presence
 
-      sign_in_logger.info('logout', @access_token.to_s)
-      StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_LOGOUT_SUCCESS)
-
-      logout_redirect = SignIn::LogoutRedirectGenerator.new(user: @current_user,
-                                                            client_config: client_config(client_id)).perform
+        SignIn::SessionRevoker.new(access_token: @access_token, anti_csrf_token:).perform
+        delete_cookies if token_cookies
+        logout_redirect = SignIn::LogoutRedirectGenerator.new(user_verification:,
+                                                              client_config: client_config(client_id)).perform
+        sign_in_logger.info('logout', @access_token.to_s)
+        StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_LOGOUT_SUCCESS)
+      else
+        user_verification = OpenStruct.new(credential_type: nil)
```

Comment:
I feel like this is a bit of a hack, and you could avoid it if you passed `credential_type` directly to `SignIn::LogoutRedirectGenerator` instead of `user_verification`.